auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/LdapRolesProvider.java

```diff
@@ -190,4 +191,25 @@ private void initialize() throws Exception {
         return roles;
     }
 
+    @Override
+    public LdapAuthHealth checkHealth() {
+
+        boolean isHealthy = true;
+        String details = null;
+        try {
+            LDAPConnection ldapConnection = connectionPool.getConnection();
```

Comment:
Look at `connectionPool.getHealthCheck()`–there are more advanced health checks we can delegate to there.